packages/compass-collection/src/components/mock-data-generator-modal/document-count-screen.tsx

```diff
@@ -0,0 +1,148 @@
+
+const titleStyles = css({
+  fontWeight: 600,
+});
+
+const descriptionStyles = css({
+  color: palette.gray.dark1,
+  fontStyle: 'italic',
+});
+
+const inputContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: spacing[600],
+  marginTop: spacing[200],
+});
+
+const estimatedDiskSizeStyles = css({
+  fontSize: '13px',
+  marginTop: spacing[100],
+});
+
+const boldStyles = css({
+  fontWeight: 600,
+});
+
+const formatBytes = (bytes: number) => {
+  const precision = bytes <= BYTE_PRECISION_THRESHOLD ? '0' : '0.0';
+  return numeral(bytes).format(precision + 'b');
+};
+
+type ErrorState =
+  | {
+      state: 'error';
+      message: string;
+    }
+  | {
+      state: 'none';
+    };
+
+interface OwnProps {
+  documentCount: number;
+  onDocumentCountChange: (documentCount: number) => void;
+}
+
+interface Props extends OwnProps {
+  schemaAnalysisState: SchemaAnalysisState;
+}
+
+const DocumentCountScreen = ({
+  documentCount,
+  onDocumentCountChange,
+  schemaAnalysisState,
+}: Props) => {
+  const estimatedDiskSize = useMemo(
+    () =>
+      schemaAnalysisState.status === 'complete' &&
+      schemaAnalysisState.schemaMetadata.avgDocumentSize
+        ? formatBytes(
+            schemaAnalysisState.schemaMetadata.avgDocumentSize * documentCount
+          )
+        : 'Not available',
+    [schemaAnalysisState, documentCount]
+  );
+
+  const isOutOfRange = documentCount < 1 || documentCount > MAX_DOCUMENT_COUNT;
+
+  const errorState: ErrorState = useMemo(() => {
+    if (isOutOfRange) {
+      return {
+        state: 'error',
+        message: `Document count must be between 1 and ${MAX_DOCUMENT_COUNT}`,
+      };
+    }
+    return {
+      state: 'none',
+    };
+  }, [isOutOfRange]);
+
+  const handleDocumentCountChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const value = parseInt(event.target.value, 10);
+    if (!isNaN(value)) {
+      onDocumentCountChange(value);
+    }
+  };
+
+  return schemaAnalysisState.status === 'complete' ? (
+    <div>
+      <Body className={titleStyles}>
+        Specify Number of Documents to Generate
+      </Body>
+      <Body className={descriptionStyles}>
+        Indicate the amount of documents you want to generate below.
+        <br />
+        Note: We have defaulted to {DEFAULT_DOCUMENT_COUNT}.
+      </Body>
+      <div className={inputContainerStyles}>
+        <TextInput
+          label="Documents to generate in current collection"
+          type="number"
+          value={documentCount.toString()}
+          onChange={handleDocumentCountChange}
+          min={1}
+          max={MAX_DOCUMENT_COUNT}
+          state={errorState.state}
+          errorMessage={
+            errorState.state === 'error' ? errorState.message : undefined
+          }
+        />
+        <div>
+          <Body className={boldStyles}>Estimated Disk Size</Body>
+          <Body className={estimatedDiskSizeStyles}>{estimatedDiskSize}</Body>
```

Comment:
[nit] the estimated size is a bit off visually, we can probably center it wrt to input field?
<img width="484" height="80" alt="Image" src="https://github.com/user-attachments/assets/688d4bb8-5ba9-4c6f-a366-093c49e9d14e" />